hoomd/hpmc/pytest/test_boxmc.py

```diff
@@ -0,0 +1,193 @@
+# This file is part of the HOOMD-blue project, released under the BSD 3-Clause
+# License.
+
+"""Test hoomd.hpmc.update.BoxMC."""
+
+import hoomd
+import pytest
+import math
+from itertools import product
+import numpy as np
+
+valid_constructor_args = [
+    dict(trigger=hoomd.trigger.Periodic(10),
+         betaP=hoomd.variant.Constant(10),
+         seed=1),
+    dict(trigger=hoomd.trigger.After(100),
+         betaP=hoomd.variant.Ramp(1, 5, 0, 100),
+         seed=4),
+    dict(trigger=hoomd.trigger.Before(100),
+         betaP=hoomd.variant.Cycle(1, 5, 0, 10, 20, 10, 15),
+         seed=87),
+    dict(trigger=hoomd.trigger.Periodic(1000),
+         betaP=hoomd.variant.Power(1, 5, 3, 0, 100),
+         seed=4),
+]
+
+valid_attrs = [
+    ('betaP', hoomd.variant.Constant(10)),
+    ('betaP', hoomd.variant.Ramp(1, 5, 0, 100)),
+    ('betaP', hoomd.variant.Cycle(1, 5, 0, 10, 20, 10, 15)),
+    ('betaP', hoomd.variant.Power(1, 5, 3, 0, 100)),
+    ('volume', {'mode': 'standard', 'weight': 0.7, 'delta':0.3}),
+    ('volume', {'mode': 'ln', 'weight': 0.1, 'delta': 1.2}),
+    ('aspect', {'weight': 0.3, 'delta': 0.1}),
+    ('length', {'weight': 0.5, 'delta': [0.8]*3}),
+    ('shear', {'weight': 0.7, 'delta': [0.3]*3, 'reduce': 0.1})
+]
+
+betaP_boxmoves = list(product([1, 3, 5, 7, 10],
+                      [ {'move':'volume', "params": {'mode':'standard', 'weight':1, 'delta':0.05}},
+                        {'move':'volume', "params": {'mode':'ln', 'weight':1, 'delta':0.05}},
+                        {'move':'aspect', "params": {'weight':1, 'delta':0.05}},
+                        {'move':'shear', "params": {'weight':1, 'delta':(0.05,)*3, 'reduce':0.2}},
+                        {'move':'length', "params": {'weight':1, 'delta':(0.05,)*3}}
+                       ]))
+
+def _is_close(v1, v2):
+    return v1 == v2 if isinstance(v1, str) else np.allclose(v1, v2)
+
+@pytest.mark.parametrize("constructor_args", valid_constructor_args)
+def test_valid_construction(constructor_args):
+    """Test that BoxMC can be constructed with valid arguments."""
+    boxmc = hoomd.hpmc.update.BoxMC(**constructor_args)
+
+    # validate the params were set properly
+    for attr, value in constructor_args.items():
+        assert getattr(boxmc, attr) == value
+
+
+@pytest.mark.parametrize("constructor_args", valid_constructor_args)
+def test_valid_construction_and_attach(simulation_factory,
+                                       two_particle_snapshot_factory,
+                                       constructor_args):
+    """Test that BoxMC can be attached with valid arguments."""
+    boxmc = hoomd.hpmc.update.BoxMC(**constructor_args)
+
+    sim = simulation_factory(two_particle_snapshot_factory())
+    sim.operations.updaters.append(boxmc)
+
+    # BoxMC requires an HPMC integrator
+    mc = hoomd.hpmc.integrate.Sphere(seed=1)
+    mc.shape['A'] = dict(diameter=1)
+    sim.operations.integrator = mc
+
+    sim.operations._schedule()
```

Comment:
```suggestion
    sim.run(0)
```
Let's use public APIs in tests where possible.